src/write.ts

```diff
@@ -401,10 +454,15 @@ async function writeBenchmarkToExternalJson(
     jsonFilePath: string,
     config: Config,
 ): Promise<Benchmark | null> {
-    const { name, maxItemsInChart } = config;
+    const { name, maxItemsInChart, autoPush } = config;
     const data = await loadDataJson(jsonFilePath);
     const prevBench = addBenchmarkToDataJson(name, bench, data, maxItemsInChart);
 
+    if (!autoPush) {
+        core.debug('Skipping storing benchmarks in external data file');
+        return prevBench;
+    }
```

Comment:
Please avoid this guard. This breaks CI and does not behave as description of `auto-push`. `auto-push` controls whether pushing commit to remote or not. It means that data should be saved to file and commit needs be created. `auto-push` option is for those who want to push the change to remote by themselves.